odoo/tools/mail.py

```diff
@@ -803,3 +818,12 @@ def parse_contact_from_email(text):
         name, email_normalized = text, ''
 
     return name, email_normalized
+
+
+def prepare_name_for_email(text):
+    if text:
+        return re.sub(
+            '([^ ]+@[^ ]+)',
```

Comment:
```suggestion
            r'[^ ]+@[^ ]+',
```
* `r` so it use the regexp colorscheme on my IDEs
* no need to make an explicit group, `group(0)` already holds "the whole string that matched"

also please compile it and make it a global variable

why is the argument `test` and not `name` btw?